stackstorm-nexus3/actions/lib/base.py

```
@@ -0,0 +1,178 @@
+                validation_errors.push("%s if missing" % field)
+                is_valid = False
+
+        return (is_valid, validation_errors)
+
+    def get_connection_defaults(self):
+        """ default properties
+
+        """
+        return {
+            "url": self.config.get('url', None),
+            "verify": self.config.get('verify', True),
+            "user": self.config.get('user', None),
+            "password": self.config.get('password', None)
+        }
+
+    def init_config(self, profile=None):
+        """ Get nexus3 server connection string
+
+        """
+        self.dial_config = self.get_connection_defaults()
+        if profile is not None:
+            profile = profile.strip()
+            if len(profile) == 0:
+                profile = None
+
+        # use 'default_profile' if 'config_profile' is empty
+        if profile is None:
+            profile = self.config.get('default_profile', None)
+
+        # Use defaults options, if neither config_profile nor default_profile is provided
+        if profile is None:
+            self.logger.info("using defaults connection values")
+        else:
+            c_profile = self.nexus_profiles.get(profile, None)
+            if c_profile is None:
+                raise MissingProfileError(
+                    "connection profile not found: '%s'" % profile)
+
+            self.dial_config.update(c_profile)
+
+        # Validate configuration
+        (is_valid, errors) = self.validate_config(self.dial_config)
+        if not is_valid:
+            raise ValidationFailError(
+                "Validation failed: %s" % errors)
+
+        return self.dial_config
+
+    def init_dialer(self):
+        try:
+            self.logger.debug("connection string: %s \n" % self.dial_config)
+            self.nexus_client = NexusClient(**self.dial_config)
+        except Exception as error:
+            self.logger.error("Couldn't instantiate nexus client %s" % error)
```

This swallows the error - is that what you want?

I would fail loudly - make sure you reraise the error.